### PR TITLE
Reorganize repo

### PR DESCRIPTION
**Pull request by gfontenot**
This reorganizes the repo a little. I've combined some files, renamed
others, and moved stuff around on disk a little. It'll probably be best
to look at this branch directly to see how the project is laid out now.

I also fixed some indentation issues and removed unnecessary code from
the test suite.

### CODE REVIEW

**Comment by tonyd256** [comment]
:+1: looks good

**Comment by Keithbsmiley** [comment]
Sounds good to me.

**Comment by gfontenot** [comment]
@tonyd256 I redid this PR. Want to take another look?

@hyperspacemark Does this reorganization help you find the things you are looking for?

File: Argo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -441,6 +411,28 @@
...
+			path = Functions;
+			sourceTree = "<group>";
+		};
+		F87EB69D1ABC5F1300E3B0AB /* Types */ = {
+			isa = PBXGroup;
+			children = (
+				F87EB69E1ABC5F1300E3B0AB /* Decoded.swift */,
+				F87EB69F1ABC5F1300E3B0AB /* JSON.swift */,
+				F87EB6A01ABC5F1300E3B0AB /* JSONDecodable.swift */,
+				F87EB6A11ABC5F1300E3B0AB /* StandardTypes.swift */,
```
Comment (tonyd256) [inline Argo.xcodeproj/project.pbxproj]: What do you think about renaming this? Standard doesn't sounds right. Maybe Native?

**Comment by tonyd256** [comment]
One small comment about a name change while we're here, but otherwise :+1:

File: Argo.xcodeproj/project.pbxproj (hunk 1 above)
Comment (gfontenot) [inline Argo.xcodeproj/project.pbxproj]: These types are part of the Standard Lib though. I could go either way.

**Pull request merged**